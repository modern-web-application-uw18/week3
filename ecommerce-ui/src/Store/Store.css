.Store {
    text-align: left;
    margin: 10px;
}

.Cart {
    display: flex;
    flex-direction: column;
    border: black 1px;
    border-style: solid;
    padding: 10px;
    width: 77%;
}

.CartItem {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    border: grey solid 1px;
    margin-bottom: 3px;
}

.Catalogue {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.Unit {
    border: black 1px;
    border-style: solid;
    padding: 10px;
    margin-bottom: 10px;
    margin-left: 10px;
    margin-right: 10px;
}

.PurchaseSection {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.CartItemImage {
    height: 50px;
    flex-grow: 1;
}

.FlexGrow1 {
    flex-grow: 1;
}

.FlexGrow2 {
    flex-grow: 2;
}

.PurchaseButton {
    width: 100px;
}
