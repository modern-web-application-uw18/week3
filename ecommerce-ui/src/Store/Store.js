import React, { Component } from 'react';
import Cart from '../Cart/Cart.js';
import Catalogue from '../Catalogue/Catalogue.js';
import './Store.css';
import PropTypes from 'prop-types';

class Store extends Component {
    constructor(props) {
        super(props);
        this.state = {
            purchasedUnits: props.initialValue
        };
    }

    onAddToCart = (unit) => {
        return () => {
            this.setState((prevState, props) => {
                const purchasedUnit = unit;
                const newPurchasedUnits = prevState.purchasedUnits;
                newPurchasedUnits.push(purchasedUnit);
                return {
                    purchasedUnits: newPurchasedUnits
                };
            });
        }
    }

    onRemoveFromCart = (idx) => {
        return () => {
            this.setState((prevState, props) => {
              const newPurchasedUnits = prevState.purchasedUnits;
              newPurchasedUnits.splice(idx, 1)
              return {
                  purchasedUnits: newPurchasedUnits
              }
            });
        }
    }
    render() {
        return (
            <div className="Store">
                <div>
                    <p>Your Cart</p>
                    <Cart 
                        itemsInCart={this.state.purchasedUnits}
                        onRemoveFromCart={this.onRemoveFromCart}
                    />
                </div>
                <div>
                    <p>Available To Book:</p>
                    <Catalogue
                        inventory={this.props.inventory}
                        onAddToCart={this.onAddToCart}
                    />
                </div>
            </div>
        );
    }
}

Store.propTypes = {
    initialValue: PropTypes.array,
    inventory: PropTypes.array
}

Store.defaultProps = {
    initialValue : []
}

export default Store;