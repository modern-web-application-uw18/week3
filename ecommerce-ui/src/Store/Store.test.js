import React from 'react';
import ReactDom from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Store from './Store.js'

describe('Store', () => {
    it('should clear smoke test for Store rendering', () => {
        const div = document.createElement('div');
        const dummyProp = [
            {
                "title": "Cardboard Box",
                "houseType": "Apartment",
                "image": "./itsABox.jpg",
                "location": {
                    "city": "Seattle",
                    "country": "USA"
                },
                "payment": {
                    "cost": 1,
                    "description": "Free cancellation"
                },
                "host": {
                    "name": "I5 underpass",
                    "isSuperhost": false
                },
                "rating": {
                    "stars": 1,
                    "reviews": 6876
                }
            }
        ];
        ReactDom.render(<Store inventory={dummyProp}/>, div);
        ReactDom.unmountComponentAtNode(div);
    });
    it('should render a minimal store front', () => {
        const dummyProp = [
            {
                "title": "Cardboard Box",
                "houseType": "Apartment",
                "image": "./itsABox.jpg",
                "location": {
                    "city": "Seattle",
                    "country": "USA"
                },
                "payment": {
                    "cost": 1,
                    "description": "Free cancellation"
                },
                "host": {
                    "name": "I5 underpass",
                    "isSuperhost": false
                },
                "rating": {
                    "stars": 1,
                    "reviews": 6876
                }
            }
        ];
        const component = TestRenderer.create(<Store inventory={dummyProp}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    
})