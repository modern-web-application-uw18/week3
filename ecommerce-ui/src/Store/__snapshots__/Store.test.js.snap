// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store should render a minimal store front 1`] = `
<div
  className="Store"
>
  <div>
    <p>
      Your Cart
    </p>
    <div
      className="Cart"
    >
      Your Cart Is Empty :(
      <p />
      <strong>
        Total: $
        0
      </strong>
    </div>
  </div>
  <div>
    <p>
      Available To Book:
    </p>
    <div
      className="Catalogue"
    >
       
      <div
        className="Unit"
      >
        <div>
          <p>
            Cardboard Box
          </p>
          <img
            alt="a house you can rent"
            src="./itsABox.jpg"
          />
          <p>
            Location: 
            Seattle
            , 
            USA
          </p>
          <div
            className="PurchaseSection"
          >
            <p>
              Cost: $
              1
            </p>
            <button
              onClick={[Function]}
            >
              Add to Cart
            </button>
          </div>
        </div>
      </div>
       
    </div>
  </div>
</div>
`;
